zkevm-circuits/src/evm_circuit/execution/begin_tx.rs

```diff
@@ -146,9 +178,6 @@ impl<F: Field> ExecutionGadget<F> for BeginTxGadget<F> {
             &mut reversion_info,
         );
 
-        // TODO: Handle creation transaction
-        // TODO: Handle precompiled
```

Comment:
I've just moved it a few lines below (so that we handle cases 1, 2, 3 and 4 as in the bus-mapping implementation:
https://github.com/scroll-tech/zkevm-circuits/blob/feat/begintx-contract-deploy/zkevm-circuits/src/evm_circuit/execution/begin_tx.rs#L393